src/js/components/TextInput/TextInput.js

```diff
@@ -360,11 +364,14 @@ const TextInput = forwardRef(
 
                   return (
                     <li
+                      id={`${id}-listbox-option-${index}`}
+                      role="option"
+                      aria-selected={selected ? 'true' : undefined}
```

Comment:
I'm thinking we should move `role="option"` and `aria-selected={selected ? 'true' : undefined}` to be on the Button instead of the li. Also not sure if we need to have the id, I wasn't seeing any difference in screenreader output with vs without it

Reply:
Moving `role="option"` to the button allows the screenreader users to select an option with ctrl-option-space

Reply:
The ID links up `aria-activedescendant` on the containing listbox, so that changes can be announced even though keyboard focus is actually in the text input. 

I’m not sure about having a list item role that then contains an option. I think the options might have to be direct children of the listbox, or if not then the default role of the `li` elements might need to be changed too. 

Great that this works in VoiceOver, though. Hopefully it’s valid ARIA too.

Reply:
If VO had better support for `activedescendant`, I think some of the live region updates could be removed too. I’m finding they sometimes lag behind the actual active option, or at other times I hear the active option announced multiple times. 

JAWS and NVDA should pick up the active option, via `aria-activedescendant`, and announce it when you navigate the list with the arrow keys.

Reply:
At Zooniverse, we use the Storybook a11y add on to test for things like invalid role nesting, or invalid role and state combinations.

I can give this change a try to see if the validator complains.